### DISCUSSION TITLE
Why Rust replaces Golang?

### DISCUSSION THREAD

**Discussion by cifer76** [jasonish/evebox#174] [General]
It looks evebox started with golang as its backend language, then switched to Rust from 2020: https://github.com/jasonish/evebox/pull/133.

May I know what's the reason to do so? we are planning a IDS product based on suricata as well and want to choose golang as the backend language, have you encountered performance issue in golang or something else?

Thanks

**Comment by jasonish** [comment]
Everything else I was working on was C and Rust, this became the odd-one out being Go, and I found it really hard to continue development of as I never really like Go (as a language), it just had some right features and was easily accessible when I needed a backend.

One day I though setout to try to port just a little bit to Rust.  It was easy, and less code that I just kept going.  This is somewhat my personal space to explore languages/frameworks, etc.

In this case I think the rewrite was worth it, as I enjoy working on it. There won't be another rewrite tho. Rust is here to stay for this project.

**Reply by cifer76** [reply]
Thanks for the detailed explanation.

I once tried Rust on a small website project but finally found myself love Go more 🤣. So it's good news for hearing you didn't switch to Rust because of performance issue of Go. Thanks again 😄
